File: 09-03-2025_Palindrome_SubStrings.cpp
class Solution {
public:
    int expandAroundCenter(const string &s, int left, int right) {
        int count = 0;
        while (left >= 0 && right < s.length() && s[left] == s[right]) {
            if (right - left + 1 >= 2) {
                count++;
            }
            left--;
            right++;
        }
        return count;
    }

    int countPS(string s) {
        int n = s.length();
        int count = 0;
        
        for (int i = 0; i < n; i++) {
            count += expandAroundCenter(s, i, i);      // Odd-length palindromes
            count += expandAroundCenter(s, i, i + 1);  // Even-length palindromes
        }
        
        return count;
    }
};

File: 09-03-2025_Palindrome_SubStrings.py
class Solution:
    
    def countPS(self, s):
        def expand_around_center(left, right):
            count = 0
            while left >= 0 and right < len(s) and s[left] == s[right]:
                if right - left + 1 >= 2:
                    count += 1
                left -= 1
                right += 1
            return count
    
        n = len(s)
        count = 0
    
        for i in range(n):
            count += expand_around_center(i, i)      # Odd-length palindromes
            count += expand_around_center(i, i + 1)  # Even-length palindromes
    
        return count
